src/Transport/Topology/Topologies/ForwardingTopology.cs

```diff
@@ -109,6 +110,14 @@ public Func<IDispatchMessages> GetDispatcherFactory()
             return () => container.Resolve<IDispatchMessages>();
         }
 
+        public StartupCheckResult ApplyPreStartupChecks()
+        {
+            var checks = container.ResolveAll<IStartupCheck>();
+            var compositeStartupChecker = new CompositeStartupCheck(checks);
```

Comment:
It was a first attempt and I didn't want to introduce many changes :) 
In my opinion we could give to users the possibilities to add "pre startup checks" extending configuration API and then we can resolve all configured checks as I did, without configuring them in the ApplyDefaults method, removing also duplicated code between StandardTopology and ForwardingTopology.

If you agree, I can elaborate on this option.